### PR TITLE
Provide more data using webhooks

### PR DESCRIPTION
**Pull request by grazor**
Based on suer/redmine_webhook

```json
{
  "payload": {
    "action": "updated",
    "issue": {
      "id": 60607,
      "subject": "Test 121",
      "description": "desctiption",
      "created_on": "2019-02-15T13:19:06.625Z",
      "updated_on": "2019-03-29T21:06:04.405Z",
      "closed_on": null,
      "root_id": 60607,
      "parent_id": null,
      "done_ratio": 0,
      "start_date": null,
      "due_date": null,
      "estimated_hours": null,
      "is_private": false,
      "lock_version": 12,
      "custom_field_values": [
        {
          "custom_field_id": 2,
          "custom_field_name": "Ветка/Branch",
          "value": ""
        },
        {
          "custom_field_id": 14,
          "custom_field_name": "Build url",
          "value": ""
        },
        {
          "custom_field_id": 21,
          "custom_field_name": "Owner",
          "value": "184"
        },
        {
          "custom_field_id": 23,
          "custom_field_name": "QA",
          "value": ""
        },
        {
          "custom_field_id": 33,
          "custom_field_name": "Priority for testing ",
          "value": ""
        }
      ],
      "project": {
        "id": 32,
        "identifier": "front",
        "name": "Frontend team",
        "description": "",
        "created_on": "2012-11-06T14:39:27.104Z",
        "homepage": ""
      },
      "status": {
        "id": 10,
        "name": "Testing"
      },
      "tracker": {
        "id": 2,
        "name": "Feature"
      },
      "priority": {
        "id": 4,
        "name": "Normal"
      },
      "author": {
        "id": 483,
        "login": "irina",
        "mail": "irina@domain.com",
        "firstname": "Irina",
        "lastname": "Lastname",
        "identity_url": null,
        "icon_url": "//www.gravatar.com/avatar/..."
      },
      "assignee": {
        "id": 257,
        "login": "ivan",
        "mail": "ivan@domain.com",
        "firstname": "Ivan",
        "lastname": "Lastname",
        "identity_url": null,
        "icon_url": "//www.gravatar.com/avatar/..."
      },
      "watchers": [
        {
          "id": 483,
          "login": "irina",
          "mail": "irina@domain.com",
          "firstname": "Irina",
          "lastname": "Lastname",
          "identity_url": null,
          "icon_url": "//www.gravatar.com/avatar/..."
        }
      ]
    },
    "journal": {
      "id": 395893,
      "notes": "",
      "created_on": "2019-03-29T21:06:04.725Z",
      "private_notes": false,
      "author": {
        "id": 754,
        "login": "sergey",
        "mail": "sergey@domain.com",
        "firstname": "Sergey",
        "lastname": "Lastname",
        "identity_url": null,
        "icon_url": "//www.gravatar.com/avatar/..."
      },
      "details": [
        {
          "id": 404625,
          "value": "10",
          "old_value": "20",
          "prop_key": "status_id",
          "property": "attr"
        },
        {
          "id": 404626,
          "value": "257",
          "old_value": null,
          "prop_key": "assigned_to_id",
          "property": "attr"
        }
      ]
    },
    "url": "http://localhost:8000/issues/60607"
  }
}
```

### CODE REVIEW

File: lib/webhook/webhook_listener.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      }.to_json
+    end
+
+    def post(request_body)
+      Thread.start do
+          begin
+              url = Setting.plugin_webhook['url']
+              if url.nil? || url == ''
+                  url = 'http://localhost:8000'
```
Comment (vanadium23) [inline lib/webhook/webhook_listener.rb]: Let's remove fallback. Instead log error and exit function

File: lib/webhook/issue_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        :root_id => @issue.root_id,
+        :parent_id => @issue.parent_id,
+        :done_ratio => @issue.done_ratio,
+        :start_date => @issue.start_date,
+        :due_date => @issue.due_date,
+        :estimated_hours => @issue.estimated_hours,
+        :is_private => @issue.is_private,
+        :lock_version => @issue.lock_version,
+        :custom_field_values => @issue.custom_field_values.collect { |value| RedmineWebhook::CustomFieldValueWrapper.new(value).to_hash },
+        :project => RedmineWebhook::ProjectWrapper.new(@issue.project).to_hash,
```
Comment (vanadium23) [inline lib/webhook/issue_wrapper.rb]: Do we need custom wrapper for related class? Why not just `to_json` for related object?

Comment (grazor) [reply lib/webhook/issue_wrapper.rb]: IMO this grants more control on data returned and allows to strip unwanted field. Diff:

With custom wrapper

```
      "project": {
        "id": 32,
        "identifier": "front",
        "name": "Frontend team",
        "description": "",
        "created_on": "2012-11-06T14:39:27.104Z",
        "homepage": ""
      },
```

Without wrapper:

```
      "project": {
        "id": 32,
        "name": "Frontend team",
        "description": "",
        "homepage": "",
        "is_public": true,
        "parent_id": 1,
        "created_on": "2012-11-06T14:39:27.104Z",
        "updated_on": "2017-04-03T13:14:05.071Z",
        "identifier": "front",
        "status": 1,
        "lft": 78,
        "rgt": 85,
        "inherit_members": false,
        "landing_page": "https://red.../projects/front/issues?query_id=123",
        "default_version_id": 644,
        "auto_issue_version": false,
        "lb_assign_role_id": null,
        "associating_cf": null,
        "activity_report_settings": null,
        "default_assigned_to_id": null
      },
```

**Comment by biozz** [comment]
Seems legit, ready to merge if no other comments.

**Comment by vanadium23** [comment]
LGTM

**Pull request merged**